nwss_wastewater/delphi_nwss/run.py

```diff
@@ -126,38 +78,52 @@ def run_module(params):
     ## compute stuff here or farm out to another function or file
-    df_pull = pull_nwss_data(socrata_token)
-    ## aggregate
-    for sensor in SIGNALS:
-        df = df_pull.copy()
-        # add weighed column
-        df = generate_weights(df, sensor)
-
-        for geo in GEOS:
-            logger.info("Generating signal and exporting to CSV", metric=sensor)
-            if geo == "nation":
-                agg_df = weighted_nation_sum(df, sensor)
-            else:
-                agg_df = weighted_state_sum(df, geo, sensor)
-            # add se, sample_size, and na codes
-            agg_df = add_needed_columns(agg_df)
-            # actual export
-            dates = create_export_csv(
-                agg_df, geo_res=geo, export_dir=export_dir, sensor=sensor
-            )
-            if len(dates) > 0:
-                run_stats.append((max(dates), len(dates)))
+    df_pull = pull_nwss_data(socrata_token, logger)
+    geomapper = GeoMapper()
+    # iterate over the providers and the normalizations that they specifically provide
+    for provider, normalization in zip(PROVIDER_NORMS["provider"], PROVIDER_NORMS["normalization"]):
+        # copy by only taking the relevant subsection
+        df_prov_norm = df_pull[(df_pull.provider == provider) & (df_pull.normalization == normalization)]
+        df_prov_norm = df_prov_norm.drop(["provider", "normalization"], axis=1)
+        for sensor in [*SIGNALS, *METRIC_SIGNALS]:
+            full_sensor_name = sensor + "_" + provider + "_" + normalization
+            df_prov_norm = df_prov_norm.rename(columns={sensor: full_sensor_name})
+            for geo in GEOS:
+                logger.info("Generating signal and exporting to CSV", metric=full_sensor_name)
+                if geo == "nation":
+                    df_prov_norm["nation"] = "us"
+                agg_df = geomapper.aggregate_by_weighted_sum(
+                    df_prov_norm,
+                    geo,
+                    full_sensor_name,
+                    "timestamp",
+                    "population_served",
+                )
+                agg_df = agg_df.rename(columns={geo: "geo_id", f"weighted_{full_sensor_name}": "val"})
+                # add se, sample_size, and na codes
+                agg_df = add_needed_columns(agg_df)
+                # actual export
+                dates = create_export_csv(agg_df, geo_res=geo, export_dir=export_dir, sensor=full_sensor_name)
+                if "archive" in params:
+                    _, common_diffs, new_files = arch_diff.diff_exports()
+                    to_archive = [f for f, diff in common_diffs.items() if diff is not None]
+                    to_archive += new_files
+                    _, fails = arch_diff.archive_exports(to_archive)
+                    succ_common_diffs = {f: diff for f, diff in common_diffs.items() if f not in fails}
+                    arch_diff.filter_exports(succ_common_diffs)
```

Comment:
issue: same thing about [the runner script](https://github.com/cmu-delphi/covidcast-indicators/blob/a6ea00355d79a47cc09db8090f6e5121c718de0f/_delphi_utils_python/delphi_utils/runner.py)

Reply:
sorry I don't follow?

Reply:
oh this was commented in the other order, I'll just delete this then. There are several endpoints which include this btw